src/MongoDB/BSON/ObjectID.cpp

```diff
@@ -44,19 +58,8 @@ void HHVM_METHOD(MongoDBBsonObjectID, __construct, const Variant &objectId)
 	} else {
 		bson_oid_init(&data->m_oid, NULL);
 	}
-}
-
-static String oidAsString(MongoDBBsonObjectIDData* data)
-{
-	String s;
-	char *data_s;
 
-	s = String(24, ReserveString);
-	data_s = s.bufferSlice().data();
-	bson_oid_to_string(&data->m_oid, data_s);
-	s.setSize(24);
-
-	return s;
+	this_->o_set(s_MongoBsonObjectID_oid, Variant(oidAsString(data)), s_MongoBsonObjectID_className);
```

Comment:
I was just asking why you keep copies of both the libbson struct and hexadecimal string. In PHPC, we only store hexadecimal string after validating it during construction (or init from `__set_state()` and `unserialize()`). We briefly create a `bson_oid_t` when serializing the PHP object to BSON.